joomla/joomla-cms
# Add collapsible sidebar

This is an attempt to bring collapsible sidebars into Joomla!
## What problem is being solved

The problem that is being solved is that of the sidebar using too much screen space while at the same time maintain the usefulness of the sidebar. This is a spin-off from the discussion in PR #4178.
## Proof of concept

Currently this PR only deals with the articles list as a proof of concept. Once we have reached a mutual agreement I will implement the needed code in all applicable views.
## How to test
1. Apply this PR and navigate to Content -> Article manager. You will see an icon above the sidebar.
   ![collapse_icon](https://cloud.githubusercontent.com/assets/359377/4094594/d8b0d01e-2fa9-11e4-92cc-df2abca666ca.png)
2. Click on the icon to collapse the sidebar
3. Close the sidebar
4. Navigate to another page
5. Go back to the articles list
6. Check that the sidebar is still collapsed
## Todo
- Test with RTL
- Disable functionality on small (phone) screen devices
## Questions
- JavaScript code is currently embedded in the articles page. Can this be moved to core.js?
- How can we check for small screen devices in PHP or JS to disable this option?
- Is this icon OK or is another icon preferred?
- Is the JavaScript code OK or can it be optimized?
## Before

![joomla_-_administration_-_article_manager_articles_-_before](https://cloud.githubusercontent.com/assets/359377/4094707/2e11b4e6-2fab-11e4-951b-ec968c3d6f0b.png)
## After

![joomla_-_administration_-_article_manager_articles_-_after](https://cloud.githubusercontent.com/assets/359377/4094712/4a2a66b4-2fab-11e4-95f6-4a1714c018aa.png)
